### DISCUSSION TITLE
How to use mitmweb to modify response header value(by using regex)?

### DISCUSSION THREAD

**Discussion by passcodex** [mitmproxy/mitmproxy#4995] [Q&A] [answered]
I only want to modify a flow response header value of the section with a regex , so I add a rule in config.yaml file, but it didn't work.

![capture_2021-12-17_15-04-58](https://user-images.githubusercontent.com/11014056/146505558-6a0b918b-5b01-4a17-9c88-c04ac167098f.png)
```
# Header modify pattern of the form "[/flow-filter]/header-
# name/[@]header-value", where the separator can be any character. The @
# allows to provide a file path that is used to read the header value
# string. An empty header-value removes existing header-name headers.
# Type sequence of str.
```
`modify_headers: [
"|~u http://bpm-m.domain.com & ~hs Location|login.xx.com|test-login.xx.com"
]` (not work)
or
`modify_headers: [
"|~u http://bpm-m.domain.com & ~hs Location|Location|login.xx.com|test-login.xx.com"
]` (full value was replaced)

Does mitmproxy(mitmweb) support modify or replace response header value with a regex?
Are there other ways to do it?

**Comment by mhils** [comment, marked as answer]
I think that's not supported yet, but something I'd be happy to see a PR for. A simple workaround would be to write your own custom addon:
```python
def response(flow):
    if "Location" in flow.response.headers:
        flow.response.headers["Location"] = flow.response.headers["Location"].replace("foo", "bar")
```

and then invoke this as an addon: https://docs.mitmproxy.org/dev/addons-overview/

**Reply by passcodex** [reply]
Thanks for your guide
